### PR TITLE
Add bitwise shift operations to AtomicRef

### PR DESCRIPTION
**Pull request by crtrott**
Doesn't look like we have unit tests yet at all for AtomicRef??

### CODE REVIEW

**Comment by crtrott** [comment]
Note: I tested this version in Kokkos in my attempt to replace the guts of View with mdspan. That is how I found these ops missing compared to what our atomic reference object did in Kokkos.

File: atomics/include/desul/atomics/Atomic_Ref.hpp [hunk 1]
```diff
@@ -79,6 +79,10 @@ class AtomicRef {
   DESUL_IMPL_DEFINE_ATOMIC_FETCH_OP(xor)
   DESUL_IMPL_DEFINE_ATOMIC_COMPOUND_ASSIGNMENT_OP(^=, xor)
   DESUL_IMPL_DEFINE_ATOMIC_FETCH_OP(nand)
```
Comment (dalg24) [inline atomics/include/desul/atomics/Atomic_Ref.hpp:81]: I think we should have them take the operand as unsigned int instead of T

**Comment by dalg24** [comment]
> Doesn't look like we have unit tests yet at all for AtomicRef?? 

I had forced all kokkos atomics to go through atomic reference. But yes you're correct.

File: atomics/include/desul/atomics/Atomic_Ref.hpp (hunk 1 above)
Comment (crtrott) [reply atomics/include/desul/atomics/Atomic_Ref.hpp:81]: isn't that true for mod too?

Comment (dalg24) [reply atomics/include/desul/atomics/Atomic_Ref.hpp:81]: I am talking about consistency with the free function signatures
https://github.com/desul/desul/blob/60c1115af231f4bfe0d70a3e0cf54b2cc0ff1594/atomics/include/desul/atomics/Fetch_Op_Generic.hpp#L57-L60
We currently define `mod` as taking the operand as a `T`.

**Comment by dalg24** [comment]
The diff rendering was fishy, so I reset on the latest version, reapplied the changes I suggested, I push forced.

**Review by trws — APPROVED** [review]
Works for me.

**Pull request merged**